src/compiler/transformers/shared/CSSinJSXTransformer.ts

```diff
@@ -0,0 +1,245 @@
+const minify = (value: string): string => compressSymbols(value.replace(/[\n]\s*/g, ''));
+
+/**
+ * @todo
+ * 1. expand the minify (way to simple :P)
+ *
+ * 2. Components as selectors
+ * 3. Minification
+ * 4. Sourcemaps
+ * 5. Minify configurable?
+ *
+ * 5. Dead Code Elimination // not needed??
+ */
+export function CSSInJSXTransformer(options?: CSSInJSXTransformerOptions): ITransformer {
+  const {
+    autoInject = true,
+    autoLabel = true,
+    cssPropOptimization = true,
+    emotionCoreAlias = '@emotion/core',
+    jsxFactory = 'jsx',
+    labelFormat = '[dirname]--[local]',
+    test = /\.(js|jsx|ts|tsx)$/,
+    // sourceMap = true
+  } = options;
+
+  const testPathRegex = path2RegexPattern(test);
+  const emotionLibraries = [[emotionCoreAlias, 'emotion'], ['@emotion/styled']];
+  // const reactLibrary = 'react';
+
+  function renderAutoLabel(): ASTNode {
+    return {
+      type: 'Literal',
+      value: `label:${labelFormat
+        .replace(/\[local\]|\[filename\]|\[dirname\]/gi, m => labelMapping[m])
+        .replace(/\-\-$/, '')};`,
+    };
+  }
+
+  return {
+    target: { test: testPathRegex },
+    commonVisitors: ({ ctx, module }) => {
+      // Keep track of imported emotion functions.
+      // Allows us to look for custom imports
+      // import { css as emotionCss } from '@emotion/core'
+      const importedEmotionFunctions = [];
+      let needsInjection = true;
+
+      // Check if we're executing an emotion call in this file
+      function isEmotionCall(node: ASTNode, index: string): boolean {
+        return (
+          // css('styles')
+          importedEmotionFunctions.indexOf(node[index].name) > -1 ||
+          // styled('obj')('styles')
+          importedEmotionFunctions.indexOf(node[index].callee && node[index].callee.name) > -1 ||
+          // styled.div('div')
+          importedEmotionFunctions.indexOf(node[index].object && node[index].object.name) > -1
+        );
+      }
+
+      const compilerJsxFactory = ctx && ctx.compilerOptions.jsxFactory;
+
+      // Process dirName and fileName only once per file
+      const filePath = module.props.fuseBoxPath.replace(/\.([^.]+)$/, '');
+      labelMapping['[dirname]'] = filePath.replace(/(\\|\/)/g, '-');
+      labelMapping['[filename]'] = filePath.replace(/(.+)(\\|\/)(.+)$/, '$3');
+
+      return {
+        onEachNode: (visit: IVisit) => {
+          const { node, parent } = visit;
+
+          switch (node.type) {
+            case 'CallExpression':
+              // append the css class label
+              if (autoLabel && !parent.callee && isEmotionCall(node, 'callee')) {
+                labelMapping['[local]'] =
+                  (parent.type === 'VariableDeclarator' && parent.id.name) ||
+                  (parent.type === 'AssignmentExpression' && parent.left.property.name) ||
+                  '';
+                node.arguments.push(renderAutoLabel());
+              }
+              break;
+
+            case 'JSXElement':
```

Comment:
what's the purpose of this block?

Reply:
Ah, that's one of the features not implemented yet.. it's called css prop optimization.

Commented out, but will be used when i start implementing missing features